spyder/api/plugins/new_api.py

```diff
@@ -572,204 +896,310 @@ def set_conf(
-                    'A spyder plugin must define a `CONF_SECTION` class '
-                    'attribute!'
+                    "A spyder plugin must define a `CONF_SECTION` class "
+                    "attribute!"
                 )
             self._conf.disable_notifications(section, option)
 
-    def restore_conf(self, option, section=None):
+    def restore_conf(
+        self,
+        option: spyder.config.types.ConfigurationKey,
+        section: str | None = None,
+    ) -> None:
         """
         Restore notifications for an option in the Spyder configuration system.
 
         Parameters
         ----------
-        option: Union[str, Tuple[str, ...]]
+        option: spyder.config.types.ConfigurationKey
             Name of the option, either a string or a tuple of strings.
-        section: str
-            Section in the configuration system.
+        section: str | None, optional
+            Name of the configuration section to use, e.g. ``"shortcuts"``.
+            If ``None``, the default, then the value of :attr:`CONF_SECTION`
+            is used.
+
+        Returns
+        -------
+        None
+
+        Raises
+        ------
+        SpyderAPIError
+            If ``section`` is not passed and :attr:`CONF_SECTION` is not
+            defined.
         """
         if self._conf is not None:
             section = self.CONF_SECTION if section is None else section
             if section is None:
                 raise SpyderAPIError(
-                    'A spyder plugin must define a `CONF_SECTION` class '
-                    'attribute!'
+                    "A spyder plugin must define a `CONF_SECTION` class "
+                    "attribute!"
                 )
             self._conf.restore_notifications(section, option)
 
     @Slot(str)
     @Slot(str, int)
-    def show_status_message(self, message, timeout=0):
+    def show_status_message(self, message: str, timeout: int = 0) -> None:
         """
-        Show message in status bar.
+        Show a message in the Spyder status bar.
 
         Parameters
         ----------
         message: str
-            Message to display in the status bar.
+            The message to display in the status bar.
         timeout: int
-            Amount of time to display the message.
+            The amount of time, in milliseconds, to display the message.
+            If ``0``, the default, the message will be shown until a plugin
+            calls :meth:`!show_status_message` again.
+
+        Returns
+        -------
+        None
         """
         self.sig_status_message_requested.emit(message, timeout)
 
-    def before_long_process(self, message):
+    def before_long_process(self, message: str) -> None:
         """
-        Show a message in main window's status bar and change the mouse
-        pointer to Qt.WaitCursor when starting a long process.
+        Perform actions required before starting a longer-running process.
```

Comment:
```suggestion
        Perform actions required before starting a long-running process.
```

`longer` than what? That makes no sense to me.